env/work/srw_python/SRWLIB_Example16.py

```diff
@@ -0,0 +1,202 @@
+                           _arUnits=['eV', 'm', 'm', 'ph/s/.1%bw/mm^2'])
+uti_plot.uti_plot2d(arII, plotMeshx, plotMeshy,
+                    ['Horizontal Position [mm]', 'Vertical Position [mm]', 'Intenisty After Propagation [a.u.]'])
+uti_plot.uti_plot1d(arI1y, plotMeshy, ['Vertical Position [mm]', 'Intensity [a.u.]',
+                                       'Intensity After Propagation\n(cut vs vertical position at x=0)'])
+
+srwl.CalcIntFromElecField(arRe, wfr, polarization, 5, 2, wfr.mesh.eStart, 0, 0)
+srwl.CalcIntFromElecField(arIm, wfr, polarization, 6, 2, wfr.mesh.eStart, 0, 0)
+
+parameters = [
+    ['Max Inten B and A propagation', arIinymax, arI1ymax],
+    ['Num of y mesh pts B and A propagation', len(arIinY), len(arI1y)],
+    ['Num of x mesh pts B and A propagation', wfrIn.mesh.nx, wfr.mesh.nx],
+    ['Num of y mesh pts B and A propagation', wfrIn.mesh.ny, wfr.mesh.ny],
+    ['Wfr xS size [mm] B and A propagation', wfrIn.mesh.xStart, wfr.mesh.xStart],
+    ['Wfr xE size [mm] B and A propagation', wfrIn.mesh.xFin, wfr.mesh.xFin],
+    ['Wfr yS size [mm] B and A propagation', wfrIn.mesh.yStart, wfr.mesh.yStart],
+    ['Wfr yE size [mm] B and A propagation', wfrIn.mesh.yFin, wfr.mesh.yFin],
+]
+for i in range(len(parameters)):
+    print('{}{}: [{}, {}]'.format('    ', parameters[i][0], parameters[i][1], parameters[i][2]))
+
+#************************************* 2. Defining parameters for analytic calculation
+lam = 2.4796e-6  # 1.2398/0.5 eV
+numPointsIn = len(arIinY)
+numPointsOut = len(arI1y)
+meshSize = float(wfr.mesh.xFin)
+
+#************************************* 3. Computing intensity distribution as per Born & Wolf, Principles of Optics
+th = []
+sIn = []
+sOut = []
+analyticalIntens = []
+for i in range(numPointsOut):
+    thx = 2.0 * (i - numPointsOut / 2.0 + 0.5) * meshSize / numPointsOut / driftLength
+    th.append(thx)
+    sOut.append(thx * driftLength * 1000)
+try:
+    from scipy.special import jv
+    for i in range(numPointsOut):
+        x = 3.1415 * apertureSize * sin(th[i]) / lam
+        analyticalIntens.append((2 * jv(1, x) / x) ** 2)
+except:
+    pass
+for i in range(numPointsIn):
+    sIn.append(2000.0 * (i - numPointsIn / 2.0) * float(wfrIn.mesh.xFin) / numPointsIn)
+
+#************************************* 4. Plotting
+try:
+    from matplotlib import pyplot as plt
+    fig = plt.figure()
+    ax = fig.add_subplot(111)
+    ax.plot(sIn, arIinY, '--g.', label='SRW (before aperture)')
+    ax.plot(sOut, arI1y, '-r.', label='SRW (after aperture)')
+    if analyticalIntens:
+        ax.plot(sOut, analyticalIntens, '-b.', label='Analytical estimation')
+    ax.legend()
+    ax.set_xlabel('Vertical Position [mm]')
+    ax.set_ylabel('Normalized Intensity, [a.u.]')
+    ax.set_title('Intensity After Propagation\n(cut vs vertical position at x=0)')
+    ax.grid()
+    plt.savefig('{}/compare.png'.format(example_folder))
+    plt.show()
+except:
+    pass
+
+print('done')
```

Comment:
Implemented. The following images are displayed after the calculations are done:
![before](https://cloud.githubusercontent.com/assets/13209176/22133501/7e6430e2-de8e-11e6-87df-91f37f754ea3.png)
![after_0m](https://cloud.githubusercontent.com/assets/13209176/22133502/81285fb0-de8e-11e6-9a77-9fbc8d1a0cae.png)
![after_1m](https://cloud.githubusercontent.com/assets/13209176/22133505/8386d822-de8e-11e6-8709-3434bc6e9451.png)
![compare](https://cloud.githubusercontent.com/assets/13209176/22133509/85b2176a-de8e-11e6-8306-48caf5667379.png)

A sample output is:
```
SRWLIB Python Example # 16:
Intensity distribution after diffraction on a circular aperture.
Comparison with an analytical distribution...
    Maximum intensity before and after propagation: [1.50472109022e+24, 9.5609834822e+22]
    Number of horizontal mesh points before and after propagation: [1120, 1152]
    Number of vertical mesh points before and after propagation: [1120, 1152]
    Wavefront horizontal start coordinates [mm] before and after propagation: [-0.01, -0.0250752705153]
    Wavefront horizontal end coordinates [mm] before and after propagation: [0.01, 0.0250752705153]
    Wavefront vertical start coordinates [mm] before and after propagation: [-0.01, -0.0250752705153]
    Wavefront vertical end coordinates [mm] before and after propagation: [0.01, 0.0250752705153]
    Vertical FWHM [mm] before and after propagation: [4.70618048306, 3.38171453018]
    Analytical FWHM [mm]: 3.40239308177
done
```
The file is on nsls2 server in `2017-01-19 SRW new developments` folder. I tried to run it  on Python 3.3 (`C:\Program Files\Python33`) which does not have scipy (in fact, scipy is not supported for 3.3 - https://github.com/scipy/scipy/blob/master/INSTALL.rst.txt#prerequisites), and it prints a message that scipy is not installed so it's not possible to perform the comparison. I hope now it's good enough for merging.